[expo-localization] Exception: Unsupported ISO 3166 country: es_419

### Minimal reproducible example

https://gist.github.com/andresribeiro/569e4d70afe1778e8f2541e301433017

### Summary

The following error appears on Play Console pre-launch report:

```
Exception com.facebook.react.common.JavascriptException: Error: Call to function 'ExpoLocalization.getLocales' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: Unsupported ISO 3166 country: es_419, js engine: hermes, stack:
_construct@1:105491
Wrapper@1:105136
_createSuperInternal@1:645189
CodedError@1:645474
findBestAvailableLanguage@1:703297
getBestAvailableLanguage@1:703599
setBestAvailableDayJsLocale@1:692514
anonymous@1:692458
loadModuleImplementation@1:73062
guardedLoadModule@1:72611
metroRequire@1:72239
anonymous@1:690154
loadModuleImplementation@1:73062
guardedLoadModule@1:72611
metroRequire@1:72239
anonymous@1:79390
loadModuleImplementation@1:73062
guardedLoadModule@1:72568
metroRequire@1:72239
global@1:71823
  at com.facebook.react.modules.core.ExceptionsManagerModule.reportException (ExceptionsManagerModule.java:72)
  at java.lang.reflect.Method.invoke
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.jni.NativeRunnable.run
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:223)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:228)
  at java.lang.Thread.run (Thread.java:923)
```

It might be related to these 2 issues from `react-native-localize`:

https://github.com/zoontek/react-native-localize/issues/31
https://github.com/zoontek/react-native-localize/issues/36

### Environment

System:
  OS: Linux 5.4 Linux Mint 20.3 (Una)
  Shell: 5.8 - /usr/bin/zsh
Binaries:
  Node: 19.7.0 - /usr/bin/node
  Yarn: 1.22.19 - /usr/bin/yarn
  npm: 9.5.0 - /usr/bin/npm
  Watchman: 20220605.192726.0 - /usr/local/bin/watchman
npmPackages:
  expo: ~48.0.4 => 48.0.4 
  react: 18.2.0 => 18.2.0 
  react-native: 0.71.3 => 0.71.3 
npmGlobalPackages:
  eas-cli: 3.7.2
  expo-cli: 6.3.2
Expo Workflow: managed

Have same issue on Expo 47:
```
Exception com.facebook.react.common.JavascriptException: Error: Call to function 'ExpoLocalization.getLocales' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: Unsupported ISO 3166 country: ar, js engine: hermes, stack:
_construct@1:153179
Wrapper@1:152824
_createSuperInternal@1:607368
CodedError@1:607653
anonymous@1:1755898
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1753927
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1753586
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1753475
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1752731
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1750784
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1750404
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1750281
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1750167
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1747243
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1746578
metroRequire@1:55984
anonymous@1:1191421
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1151148
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:1150957
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:679976
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:679750
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:669949
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:651969
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:651821
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:651679
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:651598
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:651491
loadModuleImplementation@1:56807
guardedLoadModule@1:56356
metroRequire@1:55984
anonymous@1:63189
loadModuleImplementation@1:56807
guardedLoadModule@1:56313
metroRequire@1:55984
global@1:55559
  at com.facebook.react.modules.core.ExceptionsManagerModule.reportException (ExceptionsManagerModule.java:72)
  at java.lang.reflect.Method.invoke
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.jni.NativeRunnable.run
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:237)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:228)
  at java.lang.Thread.run (Thread.java:919)
  ```

Same issue here for ar EXPO 48:

Exception com.facebook.react.common.JavascriptException: Error: Call to function 'ExpoLocalization.getLocales' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: Unsupported ISO 3166 country: ar, js engine: hermes, stack:
_construct@1:126614
Wrapper@1:126259
_createSuperInternal@1:668274
CodedError@1:668559
_handleAppStateChange@1:2231905
anonymous@1:645128
emit@1:115114
__callFunction@1:120571
anonymous@1:119079
__guard@1:120018
callFunctionReturnFlushedQueue@1:119037
  at com.facebook.react.modules.core.ExceptionsManagerModule.reportException (ExceptionsManagerModule.java:72)
  at java.lang.reflect.Method.invoke
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.jni.NativeRunnable.run
  at android.os.Handler.handleCallback (Handler.java:942)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:228)
  at java.lang.Thread.run (Thread.java:1012)
  
<img width="711" alt="image" src="https://github.com/expo/expo/assets/435573/618eb19d-2ff0-4608-b685-3fd6ecc159f9">

Getting these errors as well. With expo 49.

 Error: Call to function 'ExpoLocalization.getLocales' has been rejected. → Caused by: java.lang.IllegalArgumentException: Unsupported ISO 3166 country: en

--- a/packages/expo-localization/CHANGELOG.md
+++ b/packages/expo-localization/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- [Android] Fix es-419 locale returning empty list.
+
 ### 💡 Others
 
 ## 14.8.3 - 2024-01-18

--- a/packages/expo-localization/CHANGELOG.md
+++ b/packages/expo-localization/CHANGELOG.md
@@ -8,7 +8,7 @@
 
 ### 🐛 Bug fixes
 
-- [Android] Fix es-419 locale returning empty list.
+- [Android] Fix es-419 locale returning empty list. ([#27250](https://github.com/expo/expo/pull/27250) by [@aleqsio](https://github.com/aleqsio))
 
 ### 💡 Others
 

--- a/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
+++ b/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
@@ -137,6 +137,20 @@ class LocalizationModule : Module() {
     }
   }
 
+  private fun getCurrencyProperties(locale: Locale): Map<String, Any?> {
+    return try {
+      mapOf(
+      "currencyCode" to Currency.getInstance(locale).currencyCode,
+      // currency symbol can be localized to display locale (1st on the list) or to the locale for the currency (as done here).
+      "currencySymbol" to Currency.getInstance(locale).getSymbol(locale))
+    } catch (e: Exception) {
+      mapOf(
+        "currencyCode" to null,
+        "currencySymbol" to null
+      )
+    }
+  }
+
   private fun getPreferredLocales(): List<Map<String, Any?>> {
     val locales = mutableListOf<Map<String, Any?>>()
     val localeList: LocaleListCompat = LocaleListCompat.getDefault()
@@ -156,12 +170,8 @@ class LocalizationModule : Module() {
             "digitGroupingSeparator" to decimalFormat.groupingSeparator.toString(),
 
             "measurementSystem" to getMeasurementSystem(locale),
-            "currencyCode" to decimalFormat.currency.currencyCode,
-
-            // currency symbol can be localized to display locale (1st on the list) or to the locale for the currency (as done here).
-            "currencySymbol" to Currency.getInstance(locale).getSymbol(locale),
             "temperatureUnit" to getTemperatureUnit(locale)
-          )
+          ) + getCurrencyProperties(locale)
         )
       } catch (e: Exception) {
         // warn about the problematic locale

--- a/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
+++ b/packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt
@@ -140,9 +140,10 @@ class LocalizationModule : Module() {
   private fun getCurrencyProperties(locale: Locale): Map<String, Any?> {
     return try {
       mapOf(
-      "currencyCode" to Currency.getInstance(locale).currencyCode,
-      // currency symbol can be localized to display locale (1st on the list) or to the locale for the currency (as done here).
-      "currencySymbol" to Currency.getInstance(locale).getSymbol(locale))
+        "currencyCode" to Currency.getInstance(locale).currencyCode,
+        // currency symbol can be localized to display locale (1st on the list) or to the locale for the currency (as done here).
+        "currencySymbol" to Currency.getInstance(locale).getSymbol(locale)
+      )
     } catch (e: Exception) {
       mapOf(
         "currencyCode" to null,

